File: LAB_02/JAVA 1-2/src/AreaRectangulo.java
import java.util.Scanner;

public class AreaRectangulo {

    public static double calcularAreaRectangulo(double base, double altura) {
        if (base <= 0 || altura <= 0) {
            throw new IllegalArgumentException("La base y la altura deben ser positivas.");
        }
        return base * altura;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.println("Ingrese las dimensiones del rectángulo:");

        try {
            System.out.print("Base: ");
            double base = scanner.nextDouble();

            System.out.print("Altura: ");
            double altura = scanner.nextDouble();

            double area = calcularAreaRectangulo(base, altura);

            System.out.println("\nBase: " + base);
            System.out.println("Altura: " + altura);
            System.out.println("Área del rectángulo: " + area);

        } catch (IllegalArgumentException e) {
            System.out.println("Error: " + e.getMessage());
        } catch (Exception e) {
            System.out.println("Error: Entrada inválida.");
        } finally {
            scanner.close();
        }
    }
}

File: LAB_02/PYTHON 1-2/AreaRectangulo.py
def calcular_area_rectangulo(base, altura):
    if base <= 0 or altura <= 0:
        raise ValueError("La base y la altura deben ser positivas.")
    return base * altura

if __name__ == "__main__":
    print("Ingrese las dimensiones del rectángulo:")
    try:
        base = float(input("Base: "))
        altura = float(input("Altura: "))
        area = calcular_area_rectangulo(base, altura)
        print(f"\nBase: {base}")
        print(f"Altura: {altura}")
        print(f"Área del rectángulo: {area}")
    except ValueError as e:
        print(f"Error: {e}")